homeassistant/components/ista_ecotrend/sensor.py

```diff
@@ -0,0 +1,187 @@
+"""Sensor platform for Ista EcoTrend integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from enum import StrEnum
+import logging
+from typing import Literal
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import UnitOfEnergy, UnitOfVolume
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import IstaConfigEntry
+from .const import DOMAIN
+from .coordinator import IstaCoordinator
+from .util import get_native_value
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(kw_only=True, frozen=True)
+class IstaSensorEntityDescription(SensorEntityDescription):
+    """Ista EcoTrend Sensor Description."""
+
+    consumption_type: Literal["heating", "warmwater", "water"]
+    value_type: Literal["costs", "energy"] | None = None
+
+
+class IstaSensorEntity(StrEnum):
+    """Ista EcoTrend Entities."""
+
+    HEATING = "heating"
+    HEATING_ENERGY = "heating_energy"
+    HEATING_COST = "heating_cost"
+
+    HOT_WATER = "hot_water"
+    HOT_WATER_ENERGY = "hot_water_energy"
+    HOT_WATER_COST = "hot_water_cost"
+
+    WATER = "water"
+    WATER_COST = "water_cost"
+
+
+SENSOR_DESCRIPTIONS: dict[str, IstaSensorEntityDescription] = {
```

Comment:
```suggestion
SENSOR_DESCRIPTIONS: tuple[IstaSensorEntityDescription, ...] = (
```
I don't think there's a point in that being a dict.